src/qibolab/sweeper.py

```diff
@@ -47,10 +52,20 @@ class Sweeper:
             in case of a `ReadoutPulse` or around the drive frequency for a generic `Pulse`. If the parameter is `amplitude` the range is
             normalized with the current amplitude of the pulse. For other parameters the sweep will be performed directly over the range specified.
         pulses (list) : list of `qibolab.pulses.Pulse` to be swept (optional).
-        qubits (lilst): list of `qibolab.platforms.abstract.Qubit` to be swept (optional).
+        qubits (list): list of `qibolab.platforms.abstract.Qubit` to be swept (optional).
     """
 
     parameter: Parameter
     values: npt.NDArray
     pulses: Optional[list] = None
     qubits: Optional[list] = None
+
+    def __post_init__(self):
+        if self.pulses and self.qubits:
```

Comment:
This is a nice place for a `match` statement, unfortunately it is only available since py3.10:
https://docs.python.org/3/reference/compound_stmts.html#match
https://docs.python.org/3/tutorial/controlflow.html#match-statements

However, do not rely on truthiness/falseness of values, test explicitly `is None`.

Reply:
> This is a nice place for a `match` statement, unfortunately it is only available since py3.10: https://docs.python.org/3/reference/compound_stmts.html#match https://docs.python.org/3/tutorial/controlflow.html#match-statements

Yes, I think we need to keep support for >py3.8.

> However, do not rely on truthiness/falseness of values, test explicitly `is None`.

Fixed.

Reply:
> Yes, I think we need to keep support for >py3.8.

I know, and since a while, I start dreaming the time when we'll drop py3.9. Then `Union[int, float]` will be `int | float` (including `Optional[int]` into `int | None`) and we will use pattern matching.

However, with py3.9 (i.e. dropping py3.8) we'll already get dict/set union and `List[int]` will be `list[int]` :D